packages/react-scripts/config/webpack.config.dev.js

```diff
@@ -196,8 +194,23 @@ module.exports = {
       template: paths.appHtml,
     }),
     // Makes some environment variables available to the JS code, for example:
-    // if (process.env.NODE_ENV === 'development') { ... }. See `./env.js`.
-    new webpack.DefinePlugin(env),
+    // if (process.env.NODE_ENV === 'development') { ... }.
+    new EnvDefinePlugin({
+      // Grab REACT_APP_* environment variables
+      regex: /^REACT_APP_/i,
+      customVariables: {
+        // Useful for determining whether we’re running in production mode.
+        // Most importantly, it switches React into the correct mode.
+        'NODE_ENV': JSON.stringify(
+          process.env.NODE_ENV || 'development'
```

Comment:
We don't need to read `process.env.NODE_ENV` here at all. We don't allow overriding it. So we can just pass `'development'`.